psi4/src/psi4/libpsio/error.cc

```diff
@@ -50,89 +52,85 @@ namespace psi {
  */
 void psio_error(size_t unit, size_t errval) {
     int i;
-
     fprintf(stderr, "PSIO_ERROR: unit = %zu, errval = %zu\n", unit, errval);
     /* Try to save the TOCs for all open units */
     /* psio_tocwrite() does not call psio_error() so this is OK */
     for (i = 0; i < PSIO_MAXUNIT; i++) psio_tocwrite(i);
-
+    auto msg = new char[800];
     switch (errval) {
         case PSIO_ERROR_INIT:
-            fprintf(stderr, "PSIO_ERROR: %d (I/O inititalization failed)\n", PSIO_ERROR_INIT);
+            sprintf(msg,"PSIO_ERROR: %d (I/O inititalization failed)\n", PSIO_ERROR_INIT);
             break;
         case PSIO_ERROR_DONE:
-            fprintf(stderr, "PSIO_ERROR: %d (I/O cleanup failed)\n", PSIO_ERROR_DONE);
+            sprintf(msg,"PSIO_ERROR: %d (I/O cleanup failed)\n", PSIO_ERROR_DONE);
             break;
         case PSIO_ERROR_MAXVOL:
-            fprintf(stderr, "PSIO_ERROR: %d (maximum number of volumes exceeded)\n", PSIO_ERROR_MAXVOL);
+            sprintf(msg,"PSIO_ERROR: %d (maximum number of volumes exceeded)\n", PSIO_ERROR_MAXVOL);
             break;
         case PSIO_ERROR_NOVOLPATH:
-            fprintf(stderr, "PSIO_ERROR: %d (no volume path given)\n", PSIO_ERROR_NOVOLPATH);
+            sprintf(msg,"PSIO_ERROR: %d (no volume path given)\n", PSIO_ERROR_NOVOLPATH);
             break;
         case PSIO_ERROR_IDENTVOLPATH:
-            fprintf(stderr, "PSIO_ERROR: %d (two identical volume paths)\n", PSIO_ERROR_IDENTVOLPATH);
+            sprintf(msg,"PSIO_ERROR: %d (two identical volume paths)\n", PSIO_ERROR_IDENTVOLPATH);
             break;
         case PSIO_ERROR_OPEN:
-            fprintf(stderr, "PSIO_ERROR: %d (open call failed)\n", PSIO_ERROR_OPEN);
-            fprintf(stderr, "PSIO_ERROR:\n");
-            fprintf(stderr, "PSIO_ERROR: Check the location of your scratch directory which can be\n");
-            fprintf(stderr, "PSIO_ERROR: specified via the $PSI_SCRATCH environment variable or in\n");
-            fprintf(stderr, "PSIO_ERROR: the $HOME/.psi4rc file.\n");
-            fprintf(stderr, "PSIO_ERROR:\n");
-            fprintf(stderr, "PSIO_ERROR: Please note that the scratch directory must exist and be\n");
-            fprintf(stderr, "PSIO_ERROR: writable by Psi4\n");
+            sprintf(msg,"PSIO_ERROR: %d (open call failed)\n\n"
+                        " Check the location of your scratch directory which can be\n"
+                        " specified via the $PSI_SCRATCH environment variable or in\n"
+                        " the $HOME/.psi4rc file.\n\n"
+                        " Please note that the scratch directory must exist and be\n"
+                        " writable by Psi4\n", PSIO_ERROR_OPEN);
```

Comment:
```suggestion
            sprintf(msg,"PSIO_ERROR: %d (open call failed)\n\n"
                        " Check the location of your scratch directory which can be\n"
                        " specified via the $PSI_SCRATCH environment variable.\n"
                        " A local (non-network) scratch disk is strongly preferred.\n\n"
                        " Please note that the scratch directory must exist, be\n"
                        " writable by Psi4, and have available space.\n", PSIO_ERROR_OPEN);
```

Reply:
Done. Good catch with the `.psi4rc` file note!

Reply:
thanks! the `.psi4rc` bit is still true, afaik, but I don't want to encourage it (mostly b/c people, including myself, forget it could be lurking and influencing their calc).